modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/TracksAnalyticsTracker.kt

```diff
@@ -70,12 +38,12 @@ class TracksAnalyticsTracker @Inject constructor(
         /* Create the merged JSON Object of properties.
         Properties defined by the user have precedence over the default ones pre-defined at "event level" */
         val propertiesToJSON = JSONObject(properties)
-        predefinedEventProperties.keys.forEach { key ->
+        predefinedEventProperties.forEach { (key, value) ->
             if (propertiesToJSON.has(key)) {
                 Timber.w("The user has defined a property named: '$key' that will override the same property pre-defined at event level. This may generate unexpected behavior!!")
-                Timber.w("User value: " + propertiesToJSON.get(key).toString() + " - pre-defined value: " + predefinedEventProperties[key].toString())
+                Timber.w("User value: ${propertiesToJSON.get(key)} - pre-defined value: $value")
             } else {
-                propertiesToJSON.put(key, predefinedEventProperties[key])
```

Comment:
Using the `predefinedEventProperties` variable causes the get() to be called again.